### ISSUE TITLE
[Enhancement] Dockerize everything for dev env setup?

### ISSUE THREAD

**Issue by lpatmo** [codebuddies/django-concept#41]
## Context
A couple of people (@bengineerdavis, @d3vild06) have brought up/asked about docker-izing the entire app with docker-compose. My take was that I didn't think it was necessary, unless the benefits (easier setup) is very clear. @billglover explored it, and recorded a screencast of it here: https://codebuddies.slack.com/archives/C04BRN86J/p1578770421053800. @kennethlove offered some suggestions to tweak the approach.

## Acceptance Criteria
[ ] Explore a dockerized Django, Postgres, Mailhog, and Adminer (for viewing the DB). (Re: Mailhog -- that's fine! I expect at some point we'll be building out the transactional emails features)
[ ] Write a draft of instructions that users (like me) could test to set up the dev environment. Instructions should be friendly for those who've never used Docker before.

**Comment by billglover** [comment]
Happy to pick this up.

Have removed the documentation label as this requires the addition of a `Dockerfile` and probably a `docker-compose.yaml`.

**Comment by billglover** [comment]
For reference, this is the link to suggestions by @kennethlove: [Docker for Django](https://github.com/kennethlove/kennethlove.github.io/blob/master/content/posts/docker-for-django.md)

**Comment by billglover** [comment]
Would be nice to re-name the app from `cbv3_django_prototype` to `app`. Any thoughts on this?

**Comment by lpatmo** [comment]
Haha, that's an excellent idea. I'll file a Github ticket for it. The renaming (including the parts that need to be changed in the Dockerfile) could be all done in one go/one PR without any problems, right?

**Comment by billglover** [comment]
Yeah, probably best to do this as a separate issue.

**Comment by billglover** [comment]
Running into conflicting migrations with the latest merge. They aren't related to Docker, but I'll try to get this fixed so that we have a clean starting point.

**Comment by lpatmo** [comment]
Oof, I remember Bethany saying she ran into issues with migrations too (near the end of the thread at https://codebuddies.slack.com/archives/C04BRN86J/p1579044879187600). I didn't because I had a clean install with no data and no migration files. 

Heading to sleep now, but let me know if you need help!

**Comment by billglover** [comment]
I don't think it was that as I removed my database entirely before starting. Either way, migrations issue should be resolved and PR #44 should work. An edge case we may want to cover in documentation is that you'll need to ensure you don't have anything running that is already listening on these ports, e.g. an existing PostgreSQL install.

**Comment by lpatmo** [comment]
Thank you! Agree on making a note about the edge case. Will be able to pull
and look later tonight.

**Comment by billglover** [comment]
I believe #44 covers the above. I'd love to get more people go through testing it, but suspect that will come post merge now.

**Comment by bengineerdavis** [comment]
I'll bring up your new changes between now and tomorrow. :)
